spec/validation/val/valSpec.coffee

```diff
@@ -46,14 +30,44 @@ describe 'validation.remove_team_asset', () ->
     actual = validation.remove_team_asset(team, 'gh', {'a': 'b'})
     expect(actual).toEqual('xxx')
     expect(validation.gh.remove_team_asset).toHaveBeenCalledWith(team, {'a': 'b'})
+  it 'not allowed if the resource has no handler', () ->
+    validation.test_resource = {}
+    expect(() ->
+      actual = validation.remove_team_asset(team, 'test_resource', {'a': 'b'})
+    ).toThrow('resource, test_resource, does not support removing assets')
 
 describe 'add_team_member', ->
   it 'allowed when the role is an admin role', () ->
     actual = validation.add_team_member(team, user, 'admin')
-    expect(actual).toBe(true)
+    expect(actual).toBeUndefined()
   it 'allowed when the role is a non-admin role', () ->
     actual = validation.add_team_member(team, user, 'member')
-    expect(actual).toBe(true)
+    expect(actual).toBeUndefined()
```

Comment:
it used to be that validation returned true or false, but now that you can do complex validation, we want to be able to tell the user _why_ their input is invalid. So now, instead of returning false, we throw an error with an optional explanation. Now it doesn't matter what you return, just whether you throw an error.